src/ngMock/angular-mocks.js

```diff
@@ -1438,10 +1438,15 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
         return;
       }
     }
-    throw wasExpected ?
+    var error = wasExpected ?
         new Error('No response defined !') :
         new Error('Unexpected request: ' + method + ' ' + url + '\n' +
                   (expectation ? 'Expected ' + expectation : 'No more request expected'));
+
+    // TODO ugly!
```

Comment:
Remove the `ugly` comment 😛 
An actual comment would fine actually. Something like:

> In addition to be being converted to a rejection, this error also need to be passed to
> the `$exceptionHandler` and be rethrown (so that the test fails).